### DISCUSSION TITLE
CSS com recortes e máscaras.

### DISCUSSION THREAD

**Discussion by allefbr** [frontendbr/forum#709] [Perguntas e Respostas]
E ai galera, 

Estou com o seguinte problema, preciso implementar o layout abaixo:

![css-path](https://user-images.githubusercontent.com/2867569/30037944-2bf09542-9196-11e7-83b1-918c1cbd12d1.png)

O meu maior problema é fazer o recorte da mulher, de modo que recorte embaixo, mas não recorte em cima. 

É possível fazer isso com CSS sem precisar manipular a imagem?

**Comment by zemariobandeira** [comment]
Você pode tentar utilizar uma máscara personalizada.
Dê uma olhada nesse link: https://css-tricks.com/clipping-masking-css/
Pode ser que te ajude

**Comment by allefbr** [comment]
@MarioDKR quando estava pesquisando tinha visto esse link, porém o problema é como deixa o topo da cabela para fora do circulo, acho que não tem como fazer isso só com uma imagem dentro de uma forma.

**Comment by rubenmarcus** [comment]
vc pode ter a mulher num png transparente, e os elementos de fundo, feitos em css.. (e os por cima dela também)

**Comment by kvnol** [comment]
Eu faria exatamente o que o @rubenmarcus disse.

**Comment by allefbr** [comment]
@rubenmarcus é uma solução, porém sempre vou ter que cortar a parte de baixo da imagem no photoshop e enviar, farei assim mesmo. já que querendo ou não precisara enviar a imagem em PNG pro sistema.

Obrigado a todos 👍

**Comment by gnbezerra** [comment]
@allefbr a imagem da mulher já está com o fundo transparente, né?

Eu também faria cortando a imagem da pessoa no photoshop (se não forem dezenas dessas), mas eu tava convencido de que tinha uma forma de fazer com CSS, quase sempre tem! Aí resolvi botar a mão na massa pra descobrir e consegui.

Uma forma de fazer é com `clip-path` usando um SVG customizado como formato do recorte, conforme já é mostrado no link do @MarioDKR. No caso, seria um SVG com um formato arredondado embaixo e retangular em cima (tipo um retângulo com as bordas de baixo arredondadas, algo assim). Aí o círculo colorido fica no fundo como outro elemento.

Fiz um exemplo no Codepen de como ficaria: https://codepen.io/gnbezerra/pen/MvRELy

Na esquerda tá o resultado e na direita uma representação do SVG usado como recorte.

**Comment by allefbr** [comment]
@gnbezerra WoW, ficou muito show.

E a solução estava na minha cara o tempo todo, só não estava conseguindo visualizar o recorte.

Obrigado.
